### PR TITLE
conformance tests: use multiple AWS AZs

### PR DESCRIPTION
**Pull request by kdomanski**
**What this PR does / why we need it**:
A step towards #3456 

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/api/v1/types.go [hunk 1]
```diff
@@ -738,6 +738,10 @@ type AWSNodeSpec struct {
 	AMI string `json:"ami"`
 	// additional instance tags
 	Tags map[string]string `json:"tags"`
+	// Availiability zone in which to place the node. It is coupled with the subnet to which the node will belong.
+	AvailabilityZone string `json:"availabilityZone"`
+	// The VPC subnet to which the node shall be connected.
+	SubnetID string `json:"subnetID"`
```
Comment (mrIncompetent) [inline api/pkg/api/v1/types.go:744]: Do we already have an idea on how to handle this from a UI perspective?

Comment (kdomanski) [reply api/pkg/api/v1/types.go:744]: Yes, the list of AZ-Subnet pairs will be returned by the API and one will need to be picked fro the NodeDeployment in the UI.

Comment (mrIncompetent) [reply api/pkg/api/v1/types.go:744]: How does the API load the pairs & does the user need to select them?
Does the API simply load all available subnets?

Comment (kdomanski) [reply api/pkg/api/v1/types.go:744]: All subnets in the chosen VPC will be fetched from AWS and the user ~~chooses~~ will choose one.

**Pull request merged**